### PR TITLE
client: upgrade github.com/ugorji/go to v1.1.2

### PR DESCRIPTION
**Pull request by hnlq715**
#10477 

this PR fixes panic when updating github.com/ugorji/go to v1.1.2

maintainer note: also bumping github.com/urfave/cli to v1.20.0

### CODE REVIEW

File: client/keys.generated.go [hunk 1]
```diff
@@ -1819,20 +1915,25 @@ func (x *Node) CodecEncodeSelf(e *codec1978.Encoder) {
 					}
 				}
 				r.WriteMapStart(yynn2)
+				yynn2 = 0
```
Comment (hexfusion) [inline client/keys.generated.go]: This line needs to be manually removed `yynn2 = 0`

Comment (hexfusion) [reply client/keys.generated.go]: that will resolve this https://travis-ci.com/etcd-io/etcd/jobs/178290492#L586

**Comment by hexfusion** [comment]
Also please update you commit message per https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message

maybe `vendor: upgrade github.com/ugorji/go to v1.1.2`

**Comment by hnlq715** [comment]
@hexfusion done :-)

**Comment by hexfusion** [comment]
@hnlq715 thanks for the changes. I will review this later today/tonight.

**Comment by hexfusion** [comment]
Generally this looks good ugorji/go is now stating that codec is a module on its own this si why we are now depending on codec directly https://github.com/ugorji/go/issues/279. But I seem to be getting different output when I regenerate client so I want to better understand that. Thanks for you patience we want to make sure this is done right!

File: go.mod [hunk 2]
```diff
@@ -42,7 +42,7 @@ require (
 	github.com/spf13/pflag v1.0.1
 	github.com/stretchr/testify v1.2.2 // indirect
 	github.com/tmc/grpc-websocket-proxy v0.0.0-20170815181823-89b8d40f7ca8
-	github.com/ugorji/go v1.1.1
+	github.com/ugorji/go/codec v0.0.0-20190204201341-e444a5086c43
```
Comment (hexfusion) [inline go.mod]: For anyone curious this is valid https://github.com/ugorji/go/issues/279

**Comment by hnlq715** [comment]
Conflicts are resolved :-)

**Comment by hnlq715** [comment]
@hexfusion Any updates? :-)

**Comment by hexfusion** [comment]
To clarify we want to commits here 

`vendor: update ugorji to v1.1.2`
`client: generate new keys, manual remove yynn2 = 0`

let me know if you need help on the rebase

**Comment by hexfusion** [comment]
We are now bumping github.com/urfave/cli to v1.20.0. That wasn't part of the plan but we can use it I added the note above.

**Comment by hnlq715** [comment]
Thanks for your help @hexfusion

**Comment by xiang90** [comment]
lgtm

**Pull request merged**